pyqtgraph/widgets/MultiAxisPlotWidget.py

```diff
@@ -0,0 +1,357 @@
+            if axis.orientation in ["top", "bottom"]:
+                # connect axis main view changes to view
+                view.state["linkedViews"][view.XAxis] = weakref.ref(axis_view)
+                axis_view.sigXRangeChanged.connect(view.linkedXChanged)
+                axis_view.sigResized.connect(view.linkedXChanged)
+                # disable autorange on manual movements
+                axis_view.sigXRangeChangedManually.connect(lambda mask: self.disableAxisAutoRange(axis_name))
+            elif axis.orientation in ["right", "left"]:
+                # connect axis main view changes to view
+                view.state["linkedViews"][view.YAxis] = weakref.ref(axis_view)
+                axis_view.sigYRangeChanged.connect(view.linkedYChanged)
+                axis_view.sigResized.connect(view.linkedYChanged)
+                # disable autorange on manual movements
+                axis_view.sigYRangeChangedManually.connect(lambda mask: self.disableAxisAutoRange(axis_name))
+        view.sigStateChanged.emit(view)
+
+    def makeLayout(self, axis=None, charts=None):
+        """Adds all given axis and charts to the widget.
+
+        Parameters:
+        axis (list, None):
+            The name associated with the axis to link to whe view.
+        charts (ViewBox):
+            The ViewBox to associate the axis to.
+        """
+        self.clearLayout()
+        # SELECT AND ASSEMBLE AXIS
+        if axis is None:
+            axis = list(self.axis)
+        lo = {
+            "left": [],
+            "right": [],
+            "top": [],
+            "bottom": [],
+        }
+        for k, a in self.axis.items():
+            if k in axis:
+                lo[a.orientation].append(a)
+        vx = len(lo["left"])
+        vy = 1 + len(lo["top"])
+        # ADD TITLE ON TOP
+        self.pi.titleLabel.show()
+        self.layout.addItem(self.pi.titleLabel, 0, vx)
+        # ADD MAIN PLOTITEM
+        self.vb.show()
+        self.layout.addItem(self.vb, vy, vx)
+        # ADD AXIS
+        for x, a in enumerate(lo["left"] + [None] + lo["right"]):
+            if a is not None:
+                a.show()
+                self.layout.addItem(a, vy, x)
+        for y, a in enumerate([None] + lo["top"] + [None] + lo["bottom"]):
+            if a is not None:
+                a.show()
+                self.layout.addItem(a, y, vx)
+        # SELECT CHARTS
+        if charts is None:
+            charts = self.charts
+        last_shown = None
+        for k, c in self.charts.items():
+            c.plotItem.vb.state["isTopLevel"] = False
+            try:
+                c.plotItem.vb.sigMouseDragged.disconnect()
+            except (TypeError, RuntimeError):
+                pass
+            try:
+                c.plotItem.vb.sigMouseWheel.disconnect()
+            except (TypeError, RuntimeError):
+                pass
+            try:
+                c.plotItem.vb.sigHistoryChanged.disconnect()
```

Comment:
These disconnects here seem fragile. If ViewBox changes how it handles signals, or other places expect the viewbox signals to behave a certain way, nobody will know to check here to make sure they're being disabled properly. I'm not exactly sure how best to protect against that, though. At a minimum, I think this code should be moved onto the ViewBox class, e.g. `ViewBox.disableInteractivity()`, or combine it with the connection code below and make it something like `ViewBox.deferInteractivitySignalsTo(other_view_box)`. I can also imagine different structures of objects that would be better able to represent the relationship of nested viewboxes (e.g. explicit nesting when initializing), but I would have to play with some of those implementations more to know if they were legitimately better.

Reply:
One "easy" way to check that we're successfully disconnecting is make sure we get a `True` return on the `disconnect` call (which actually returns a `bool`).  As @0000matteo0000 noticed, pyside bindings are picky with their disconnect calls.  I did find a way we can call disconnect in a way everyone is happy, and that's by storing the connect meta object...

```python
# in __init__
self.metaObjs = {}

# when connecting
self.metaObsj['sigHistoryChanged'] = c.plotItem.vb.sigHistoryChanged.connect(some_slot)

# when disconnecting
successfulDisconnect = self.disconnect(self.metaObjs['sigHistoryChanged']
```

Reply:
> One "easy" way to check that we're successfully disconnecting is make sure we get a `True` return on the `disconnect` call (which actually returns a `bool`). As @0000matteo0000 noticed, pyside bindings are picky with their disconnect calls. I did find a way we can call disconnect in a way everyone is happy, and that's by storing the connect meta object...
> 
> ```python
> # in __init__
> self.metaObjs = {}
> 
> # when connecting
> self.metaObsj['sigHistoryChanged'] = c.plotItem.vb.sigHistoryChanged.connect(some_slot)
> 
> # when disconnecting
> successfulDisconnect = self.disconnect(self.metaObjs['sigHistoryChanged']
> ```

I just did this! Thanks for the suggestion though!